sklbench/benchmarks/common.py

```diff
@@ -39,4 +39,16 @@ def main_template(main_method):
     filters = json.loads(args.filters)["filters"]
 
     results = main_method(bench_case, filters)
-    print(json.dumps(results, indent=4))
+    
+    # only print results for rank 0
+    try:
+        from mpi4py import MPI
+        comm = MPI.COMM_WORLD
+        rank = comm.Get_rank()
+        size = comm.Get_size()
+        if rank > 0:
+            return
+    except:
+        print(json.dumps(results, indent=4))
+    else:
+        print(json.dumps(results, indent=4))
```

Comment:
This block of code changes function from somewhat abstract to specific.

Reply:
When running in distributed mode the same json is printed once on every device, yielding a result that is seemingly not json compatible. I added in logic to only print on a single device so results is json compatible (results are generally the same across devices). Any suggested alternative options?